tests/unittests/api/helpers/test_storage.py

```diff
@@ -0,0 +1,55 @@
+"""Test file for storage functions."""
+import os
+from urllib.parse import urljoin, urlparse
+
+from flask import request
+
+from app import current_app as app
+from app.api.helpers.storage import generate_hash, UploadedFile, upload_local
+from tests.unittests.setup_database import Setup
+from tests.unittests.utils import OpenEventTestCase
+
+
+class TestStorageHelperValidation(OpenEventTestCase):
+    """Test class for testing storage helper functions."""
+
+    def setUp(self):
+        """Method to set up app for testing."""
+        self.app = Setup.create_app()
+
+    def test_upload_local(self):
+        """Method to test the upload_local function."""
+        test_name = 'image.png'
+        test_path = os.path.join(app.config['BASE_DIR'],
+                                 'static/media/temp',
+                                 test_name
+                                 )
+        with open(test_path, 'wb') as img_file:
+            img_file.write(b'test content')
+
+        upfile = UploadedFile(test_path, 'image.png')
+        test_key = 'test_key'
+
+        with app.test_request_context():
+            test_hash = generate_hash(test_key)
+            test_url = request.url
+            upload_url = upload_local(upfile, key=test_key)
+
+        expected_url = urljoin(
+            '{url.scheme}://{url.netloc}'.format(url=urlparse(test_url)),
+            os.path.join('static/media/',
+            test_key,
+            test_hash,
+            test_name)
+            )
+
+        self.assertEqual(upload_url, expected_url)
```

Comment:
@iamareebjamal Am I going in the right direction now?

Reply:
You don't need app context and all of this to test that feature. This is an integration test, they are expensive, slow and not needed to test a simple function.

Move that logic in a new function, let's say `create_url` with parameters request_url and file_relative_path

```
url = urlparse(request.url)

# No need to specify scheme-corresponding port
port = url.port
if port and url.scheme == SCHEMES.get(url.port, None):
    port = None

return '{scheme}://{hostname}:{port}{file_relative_path}'.format(
       scheme=url.scheme, hostname=url.hostname, port=port,
       file_relative_path=file_relative_path).replace(':None', '')
```

Just call this method and test the output string